### DISCUSSION TITLE
Deploy to digital ocean without moving DNS servers.

### DISCUSSION THREAD

**Discussion by rdmorris** [digitalocean/supabase-on-do#17] [Q&A]
I would like to use this repo to deploy supabase to a DO Droplet however I am unable to move the DNS servers away from cloudflare. What do I need to change in the packer/terraform configs to accomplish this and later create the A record in cloudflare?

**Comment by timothymamo** [comment]
Hey @rdmorris ,

I haven't used Cloudflare myself but you'd have to change the following [resource](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/network.tf#L22) with a [cloudflare_record](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/record). To do this you would also have to add the cloudflare provider and appropriate tokens/variables in the [provider file](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/provider.tf).

You will also have to change all the records for sendgrid ([domain_auth_0](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/sendgrid.tf#L37), [domain_auth_1](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/sendgrid.tf#L46), [domain_auth_2](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/sendgrid.tf#L55), [link_brand_0](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/sendgrid.tf#L64), [link_brand_1](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/sendgrid.tf#L72)) with the appropriate `cloudflare_record`.

So a fair few changes. What's the reason you say you are `unable to move the DNS servers away from cloudflare`?

**Reply by rdmorris** [reply]
Well it's a tragic tale of an employee who forgot to back up their 2fa details before resetting their phone and permanently locking themselves out of godaddy.com who basically wants us to sue them for access to prove we are the account holders.

**Reply by timothymamo** [reply]
Ahhh yes, then changing the above resources is definitely easier. More than happy to help out further if needed, but for now hope the above helps.
